src/Main.hs

```diff
@@ -9,27 +9,28 @@ import HigherLower.Game
 main :: IO ()
 main = do
   g <- createGame
---  runState (mfix play) $ asStateTIO g
-  return ()
+  evalStateT playGame g
   
 createGame :: IO Game
 createGame = do
   low <- retryPrompt "Not a number" (promptInt "Give a lower bound")
   high <- retryPrompt "Not a number" (promptInt "Give an upper bound")
   turns <- retryPrompt "Not a number" (promptInt "How many tries do you get?")
   secret <- getStdRandom (randomR (low, high))
-  return (Game (low, high) turns (Active turns))
-
-play :: StateT Game IO () -> StateT Game IO ()
-play state = do
-  i <- asStateTIO $ retryPrompt "Not a number" (promptInt "How many tries do you get?")
-  result <- guess i
-  
-  return ()
-
-asStateTIO :: IO a -> StateT Game IO a
-asStateTIO = liftIO
+  return (Game (low, high) secret (Active turns))
```

Comment:
Making the secret really dynamic. It was accidentely using `turns` as the secret. Maybe to easy debugging?